test_pull_requests

```diff
@@ -1214,22 +1453,22 @@ popd
           Properties['settings'].each_value do |s|
             if s['pretest_settings_key'] && merge_pretest_success
               if Properties['settings'][s['pretest_settings_key']]['name'] == settings['name']
-                if comment =~ /^#{settings['test_prefix']} SUCCESS \(([^\)]+)\)/
+                if fields[:state] == :success
                   build_url = $1
                   trusted_trigger_time, trigger_login = get_trusted_trigger_time(req, comments, s)
                   if trusted_trigger_time
                     begin
                       repo_to_pull_request.each do |repo, pull_request|
                         test_merge_pull_request(pull_request['number'], repo, s)
-                      end if repo_to_pull_request.length > 0
```

Comment:
@dobbymoodge It seems right although not a necessary to be there at all afaict.  But can't you move it all the way up above 

Properties['settings'].each_value do |s|

and unless .empty?  would be better.